library/java/io/envoyproxy/envoymobile/engine/AndroidNetworkMonitor.java

```diff
@@ -39,11 +43,22 @@ public static void load(Context context) {
       if (instance != null) {
         return;
       }
-      instance = new AndroidNetworkMonitor(context);
+      instance = new AndroidNetworkMonitor(context, envoyEngine);
     }
   }
 
-  private AndroidNetworkMonitor(Context context) {
+  private AndroidNetworkMonitor(Context context, EnvoyEngine envoyEngine) {
+    int permission =
+        ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_NETWORK_STATE);
+    if (permission == PackageManager.PERMISSION_DENIED) {
+      try {
+        envoyEngine.recordCounterInc(PERMISSION_DENIED_STATS_ELEMENT, 1);
```

Comment:
From a conversation with @goaway and @junr03, the suggestion was to add a stat for tracking when a user has their permissions denied. This is a bit stranger of a usage area because this class is actually used within the various `*EngineImpl` so we're kind of forced to use the primitive `EnvoyEngine` which exposes the primitive stats methods.

cc @jingwei99